Backend/DatabaseRequester.py

```diff
@@ -1,60 +1,67 @@
 import psycopg2
-from configparser import ConfigParser
+from psycopg2 import pool
 
-# TODO : Connection pooling
-# TODO : fix error with fetchall() returning none
+import DataStructures
 
 
 class DatabaseRequester:
     # get database info from direct arguments
     def __init__(self, host, port, database, user, password):
-        self.conn = None
-        self.connected = False
         self.host = host
         self.port = port
         self.database = database
         self.user = user
         self.password = password
 
-    def connect(self):
-        if self.conn is None:
-            try:
-                self.conn = psycopg2.connect(
+    def query(self, query_string, *argv):
+        data = None
+        try:
+            status = None
+            conn = psycopg2.connect(
                     host=self.host,
                     port=self.port,
                     database=self.database,
                     user=self.user,
-                    password=self.password
-                )
-                self.connected = True
-                print("DB Connected")
-            except psycopg2.DatabaseError as err:
-                print("[ERROR] Database Connection Failed\n", err)
-                return
-
-    def disconnect(self):
-        if self.conn is not None:
-            self.conn.close()
-            self.conn = None
-            self.connected = False
-            print("DB Closed")
+                    password=self.password)
+            cur = conn.cursor()
+            cur.execute(query_string, argv)
+            conn.commit()
+
+            status = cur.statusmessage
+
+            if cur is not None:
+                # create return object data if query is SELECT
+                if "SELECT" in status:
+                    colnames = [desc[0] for desc in cur.description] # name of each column
+                    rows = cur.fetchall()  # all the rows returned
+                    data = QueryTuples(colnames, rows)
+
```

Comment:
Design Document II said that this would return "a string (“INSERT”, “DELETE”, or “UPDATE”) and integer values representing the number of rows affected" if the query was an INSERT/DELETE/UPDATE. Would you please either implement that or just put a TODO comment in the code here?

Reply:
Sorry to be a bother, but the documentation says that if the query isn't a SELECT then it'll return the number of rows affected. This can easily be retrieved using psycopg2's cursor object's rowcount field. Would you mind adding that to the QueryTuples class?